File: 심세현/BOJ/step_7/String1157.java

package beckjoon.string;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class String1157 {
        public static void main(String[] args) throws IOException {
            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
            String input = br.readLine().toUpperCase();
            int[] arr = new int[26];

            for(int i=0; i<input.length(); i++){
                arr[(int)input.charAt(i) - 65]++;
            }
            int max = -1;
            char ch = '?';

            for(int i=0; i<arr.length; i++){
                if(max < arr[i]){
                    max = arr[i];
                    ch = (char)(i+65);
                }else if(max == arr[i]){
                    ch = '?';
                }
            }
            System.out.println(ch);
    }
}

File: 심세현/BOJ/step_6/string_1157.kt
package step

import java.util.*

private val br = System.`in`.bufferedReader()

fun main() {
    var str = br.readLine().uppercase().toCharArray()
    var arr = Array<Int>(26, {0})

    for(i in 0 until str.size){
        arr[str[i].toInt() - 65]++
    }
    var max = -1
    var ch = '?'

    for(i in 0 until arr.size){
       if(max < arr[i]){
           max = arr[i]
           ch = (i+65).toChar()
       }
       else if(max == arr[i]) ch = '?'
    }
    println(ch)
}